### ISSUE TITLE
Bug: When grouping only first 200 entities are fetched

### ISSUE THREAD

**Issue by rkrug** [ropensci/openalexR#207]
When grouping, `oa_request()` only returns the first 200 groups, even if there are more More can be retrieved through paging. I tried disabling the lines

https://github.com/ropensci/openalexR/blob/52aed8748746b96b278d0cb1004cf7ff64ec5c11/R/oa_fetch.R#L318-L320

but this seems to download the whole dataset.

So:

``` r
library(openalexR)
#> Thank you for using openalexR!
#> To acknowledge our work, please cite the package by calling `citation("openalexR")`.
#> To suppress this message, add `openalexR.message = suppressed` to your .Renviron file.
q <- openalexR::oa_query(search = "biodiversity", group_by = "primary_topic.id")

## Open the folloeing URL in your browser to see the results
paste0(q, "&cursor=*")
#> [1] "https://api.openalex.org/works?search=biodiversity&group_by=primary_topic.id&cursor=*"

x <- q |> oa_request(verbose = TRUE)

length(x)
#> [1] 200
```

<sup>Created on 2024-02-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by trangdata** [comment]
Thank you for reporting @rkrug. 

When I first encountered this group_by issue, I was expecting that meta.count returns the total number of items across all pages, not just the first page. However, OpenAlex team said they were unable to report the total number of groups in meta.count due to technical limitations.

They now report `groups_count`, the number of groups **on that page**. For example, the last page of your request, page 20, shows a `groups_count` of 145.
https://api.openalex.org/works?search=biodiversity&group_by=primary_topic.id&cursor=Imh0dHBzOi8vb3BlbmFsZXgub3JnL1QxNDM0NyI=

```json
{
    "meta": {
        "count": 651701,
        "db_response_time_ms": 197,
        "page": null,
        "per_page": 200,
        "next_cursor": "Imh0dHBzOi8vb3BlbmFsZXgub3JnL1QxNDUyMSI=",
        "groups_count": 145
    },
```

I was in the middle of a fix but just now finished it in #184. Now, you can do:

``` r
library(openalexR)
out <- oa_fetch(
  search = "biodiversity",
  group_by = "primary_topic.id",
  verbose = TRUE
)
#> Requesting url: https://api.openalex.org/works?search=biodiversity&group_by=primary_topic.id
#> 
#> Downloading groups...
#> |=====================
head(out)
#>                           key
#> 1 https://openalex.org/T10001
#> 2 https://openalex.org/T10002
#> 3 https://openalex.org/T10003
#> 4 https://openalex.org/T10004
#> 5 https://openalex.org/T10005
#> 6 https://openalex.org/T10006
#>                                          key_display_name count
#> 1      Tectonic and Geochronological Evolution of Orogens   158
#> 2               Advancements in Density Functional Theory     5
#> 3      Knowledge Management and Organizational Innovation    75
#> 4 Soil Carbon Dynamics and Nutrient Cycling in Ecosystems  4981
#> 5      Biodiversity Conservation and Ecosystem Management 23776
#> 6             Work Engagement and Organizational Behavior     3
```

<sup>Created on 2024-02-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by rkrug** [comment]
Thanks @trangdata for the swift fix - very much appreciated. I installed the `coro` branch and it is now working as expected.
I will put it immediately in a report which we need next week!
